### DISCUSSION TITLE
What does $ do in Assembly?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#116447] [Programming Help]
### Body

I know stack manipulation, register manipulation, data types, conditional jumps, jumps, comparing, adding, subtracting, multiplying, calling, dividing and moving data around in X86 Assembly language. Do I just let it go to waste or do I learn more and figure out what to do with it?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Xhst** [comment]
The exact meaning of $ may vary depending on the syntax of the assembly language or assembler used.
In x86 assembly language, the $ symbol typically represents the program counter (the current address or the location of the current instruction being assembled), it's primarily used for generating position-independent code or for calculating relative offsets.

```
mov eax, variable - $
```
This instruction calculates the offset between the current address and the address of the label "variable" and stores it in the eax register.

```
jmp $
```
This instruction  jumps to the current address (creating an infinite loop).

**Reply by ghost** [reply]
```assembly
variable: equ $ - anotherVariable
```

**Reply by Xhst** [reply]
This line defines the variable as equal to the difference between the current address $ and the address of anotherVariable, calculates the offset between the current address and the address of anotherVariable.
